airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/jsonl_parser.py

```diff
@@ -2,24 +2,61 @@
 # Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 #
 
+import json
+import logging
 from typing import Any, Dict, Iterable
 
 from airbyte_cdk.sources.file_based.config.file_based_stream_config import FileBasedStreamConfig
 from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader
 from airbyte_cdk.sources.file_based.file_types.file_type_parser import FileTypeParser
 from airbyte_cdk.sources.file_based.remote_file import RemoteFile
+from airbyte_cdk.sources.file_based.schema_helpers import PYTHON_TYPE_MAPPING, merge_schemas
 
 
 class JsonlParser(FileTypeParser):
 
     MAX_BYTES_PER_FILE_FOR_SCHEMA_INFERENCE = 1_000_000
```

Comment:
why do we have this instead of tweaking the number of files to use for schema inference?

Reply:
We do both. We check bytes here but not in the other parsers because we're reading many rows to get the schema for JSONL files and want to be able to cut it off at some point for very large files.